ReferenceRequestDetailAPI

ChrisCurrin opened the pull request:
- add repo for applicationModel to get form, sections, question
- add tests for repo for applicationModel
- update repo for response to include 
  - get response by user_id
  - get answers by response_id
  - get answers by question_id 
  - get answer by response_id and by question_id
  - getting [(Question, Answer)] by section_id and response_id 
- added tests for updated response repo
- reverted response model from previous commit to adjust PR 624 based on feedback (exclude candidate_title, etc.) 
- update testing to add multiple dummy users at once from a list of names

ChrisCurrin commented:
tests were working locally, will check them out next week (going away for the weekend) and revert back

avishkar58 commented on api/app/responses/tests.py:
```diff
@@ -45,43 +57,66 @@ def _seed_data(self):
         self.test_answer1 = Answer(
-            self.test_response.id, self.test_question.id, 'My Answer')
+                self.test_response.id, self.test_question.id, 'My Answer')
         _add_object_to_db(self.test_answer1)
```
Can you update these to use your self.add_to_db function instead and remove the _add_object_to_db function

avishkar58 commented on api/app/utils/testing.py:
```diff
@@ -19,13 +24,51 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
+    try:
+        text = unicode(text, 'utf-8')
+    except (TypeError, NameError): # unicode is a default on python 3
+        pass
```
`six.ensure_text` ?

ChrisCurrin replied:
good suggestion, thanks!

avishkar58 commented:
Hey @ChrisCurrin - did you try test this locally using docker? Was there some step you needed to do to get it to mount the names.txt file? It also fails on my machine with the same error as on CircleCI...

avishkar58 commented:
Updated the path and the tests are working now.

ChrisCurrin commented:
sorry about the mess with the merges 🙈

> I accidentally merged feature branch into local develop, but "undoing" and reverting didn't fix it, so after a series of merges (and cherry-picking changes), there should be parity with the initial feature implementation and the latest changes to develop

and the delay of course! I'll be faster with addressing comments.
